python/python/Ice/Future.py

```diff
@@ -346,4 +513,8 @@ def _warn(self, msg):
     StateDone = "done"
 
 
-__all__ = ["Future", "wrap_future", "FutureBase"]
+# FutureLike is a type alias for an asyncio.Future, concurrent.futures.Future, or any awaitable object.
+FutureLike: TypeAlias = asyncio.Future[_T] | concurrent.futures.Future[_T] | FutureBase[_T]
```

Comment:
We can update this in the future to include other Future types, or a protocol (FutureProtocol) that can match some hopefully general future types.

Reply:
I don't think we need to have such a broad Future definition. If you look at `Dispatch.py` the only place we use it, the requirements for the interface are much more narrow. Just need the `add done_callback` and `result` methods.